pkg/apiserver/handlers_test.go

```diff
@@ -151,6 +151,15 @@ func TestGetAPIRequestInfo(t *testing.T) {
 		{"GET", "/namespaces", "list", "", "", "namespaces", "", "Namespace", "", []string{"namespaces"}},
 		{"GET", "/namespaces/other", "get", "", "other", "namespaces", "", "Namespace", "other", []string{"namespaces", "other"}},
```

Comment:
This test is passing, and I added the first two tests... doesn't look like GetAPIRequestInfo correctly handled namespace subresources, so I fixed that and updated the namespace subresource tests for `status` and `finalize`

Looking at it though, I think the changes in `BeforeUpdate` will cause problems for namespaces... will dig there